ml/src/main/scala/ai/h2o/sparkling/ml/params/NullableBigDataFrameParam.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.params
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, ObjectInputStream, ObjectOutputStream}
+import java.util.Base64
+
+import ai.h2o.sparkling.utils.ScalaUtils.withResource
+import org.apache.spark.ml.param.{Param, Params}
+import org.apache.spark.sql.{DataFrame, Row}
+import org.json4s.JsonAST.{JNull, JString}
+import org.json4s.jackson.JsonMethods.{compact, parse, render}
+
+// This type of Dataframe parameter just serialize execution plans, not data.
```

Comment:
What is the motivation behind that?

I was thinking - once the pipeline is fitted, the reference to the leaderboard frame can be released. And serializing an unfitted pipeline with a defined leaderboard frame doesn't really make sense to me.

Please take this as me thinking out loud.

Reply:
> once the pipeline is fitted, the reference to the leaderboard frame can be released.

Yes, that's correct. If  the pipeline has been trained, the `PipelineModel` contains just `H2OMOJOModel`, which doesn't hold `leaderboard_frame`.


This class is needed in two cases:
- In Spark you can save and load unfitted pipeline. ([This test](https://github.com/h2oai/sparkling-water/pull/2505/files#diff-cbc1bc71745ce2fce921cacdcd297c07c410ff66f7a6dff978c22c6f10b13205R139) demonstrates the scenario)
- A parameter is also serialized during the communication between pyspark and scala backend

Reply:
> A parameter is also serialized during the communication between pyspark and scala backend
This is not true, it's serialized only during the serialization of unfitted pipeline.

I suggest to remove this class in the bellow PR:
https://github.com/h2oai/sparkling-water/pull/2516